### ISSUE TITLE
How to implement succesor and addition functions?

### ISSUE THREAD

**Issue by ELC** [pythological/kanren#69]
Is there a way to implement the `successor` function and the `add` function using pure relations and kanren?

**Comment by brandonwillard** [comment]
Yes, here's one example: https://github.com/TheReasonedSchemer2ndEd/CodeFromTheReasonedSchemer2ndEd/blob/master/trs2-arith.scm
